src/include/sof/lib/pm_runtime.h

```diff
@@ -47,8 +47,30 @@ enum pm_runtime_context {
 struct pm_runtime_data {
 	spinlock_t lock;	/**< lock mechanism */
 	void *platform_data;	/**< platform specific data */
+#if CONFIG_POWER_STATE_COUNTERS
+	struct p_counters_data *p_counters; /**< debug power state counters */
+#endif
 };
 
+#if CONFIG_POWER_STATE_COUNTERS
+/**
+ * \brief Power states
+ * R0, R1, R2 are abstract power states which can be used differently
+ * based on platform implementation
+ */
+enum dsp_p_state {
+	r0_p_state = 0,
+	r1_p_state,
+	r2_p_state
+};
+
+/** \brief Debug counters data */
```

Comment:
Pls limit use of word "debug" in a code that does not contain extra features turned on for "debug" builds. Could be replaced with "diagnostic" (or some other word that better indicates production-enabled feature).